code/go/0chain.net/chaincore/httpclientutil/http_client_util_test.go

```diff
@@ -0,0 +1,1030 @@
+					t.Fatal(err)
+				}
+			},
+		),
+	)
+	defer server.Close()
+
+	errServer := httptest.NewServer(
+		http.HandlerFunc(
+			func(rw http.ResponseWriter, r *http.Request) {
+				rw.WriteHeader(http.StatusInternalServerError)
+			},
+		),
+	)
+	defer errServer.Close()
+
+	type args struct {
+		remoteUrl string
+		result    interface{}
+	}
+	tests := []struct {
+		name    string
+		args    args
+		wantErr bool
+	}{
+		{
+			name: "Request_Creating_ERR",
+			args: args{
+				remoteUrl: string(rune(0x7f)),
+			},
+			wantErr: true,
+		},
+		{
+			name: "Client_ERR",
+			args: args{
+				remoteUrl: "/",
+			},
+			wantErr: true,
+		},
+		{
+			name: "Resp_Status_Not_Ok_ERR",
+			args: args{
+				remoteUrl: errServer.URL,
+			},
+			wantErr: true,
+		},
+		{
+			name: "JSON_Decoding_ERR",
+			args: args{
+				remoteUrl: server.URL,
+				result:    "}{",
+			},
+			wantErr: true,
+		},
+		{
+			name: "OK",
+			args: args{
+				remoteUrl: server.URL,
+				result:    &map[string]interface{}{},
+			},
+			wantErr: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if err := MakeGetRequest(tt.args.remoteUrl, tt.args.result); (err != nil) != tt.wantErr {
+				t.Errorf("MakeGetRequest() error = %v, wantErr %v", err, tt.wantErr)
+			}
+		})
+	}
+}
+
+func TestMakeClientBalanceRequest(t *testing.T) {
+	balance := state.Balance(5)
+	server := httptest.NewServer(
+		http.HandlerFunc(
+			func(rw http.ResponseWriter, r *http.Request) {
+				st := &state.State{
+					Balance: balance,
+				}
+				blob, err := json.Marshal(st)
+				if err != nil {
+					t.Fatal(err)
+				}
+
+				if _, err := rw.Write(blob); err != nil {
+					t.Fatal(err)
+				}
+			},
+		),
+	)
+	defer server.Close()
+
+	invServer := httptest.NewServer(
+		http.HandlerFunc(
+			func(rw http.ResponseWriter, r *http.Request) {
+				if _, err := rw.Write([]byte("}{")); err != nil {
+					t.Fatal(err)
+				}
+			},
+		),
+	)
+	defer invServer.Close()
+
+	errServer := httptest.NewServer(
+		http.HandlerFunc(
+			func(rw http.ResponseWriter, r *http.Request) {
+				rw.WriteHeader(http.StatusInternalServerError)
+			},
+		),
+	)
+	defer invServer.Close()
+
+	type args struct {
+		clientID  string
+		urls      []string
+		consensus int
+	}
+	tests := []struct {
+		name    string
+		args    args
+		want    state.Balance
+		wantErr bool
+	}{
+		{
+			name: "ERR",
+			args: args{
+				urls: []string{
+					"worng url",
+					errServer.URL,
+					invServer.URL,
+				},
+			},
+			wantErr: true,
+		},
+		{
+			name: "Empty_ERR",
+			args: args{
+				urls: []string{},
+			},
+			wantErr: true,
+		},
+		{
+			name: "Consensus_ERR",
+			args: args{
+				urls: []string{
+					server.URL,
+				},
+				consensus: 200,
+			},
+			wantErr: true,
+		},
+		{
+			name: "OK",
+			args: args{
+				urls: []string{
+					server.URL,
+				},
+				consensus: 0,
+			},
+			want: 5,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
```

Comment:
Closure error, missing `tt:=tt`.

Reply:
There is no race between goroutines. Test running order is determined by tests struct slice.

Reply:
Fixed.